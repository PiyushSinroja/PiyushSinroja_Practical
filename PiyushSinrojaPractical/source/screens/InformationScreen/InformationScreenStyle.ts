import {StyleSheet} from 'react-native';
import {globalColors} from '../../constant/Colors';

/**
 * Information Screen Styles
 */
export const InformationScreenStyle = StyleSheet.create({
  safeAreaStyle: {
    flex: 1,
  },
  container: {
    backgroundColor: 'white',
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'flex-end',
  },
  text: {
    color: globalColors.black,
    fontSize: 19,
    marginTop: 10,
    textAlign: 'left',
    height: 25,
  },
  userText: {
    marginLeft: 10,
    marginRight: 15,
  },
});
