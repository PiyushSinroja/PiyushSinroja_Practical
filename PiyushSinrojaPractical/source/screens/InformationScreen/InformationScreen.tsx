import React, {useEffect} from 'react';
import {Text, View, SafeAreaView} from 'react-native';
import {useSelector} from 'react-redux';
import {InformationScreenStyle} from './InformationScreenStyle';

/**
 * Infromation screen
 *  used to display user information
 * @returns
 */
const InformationScreen: React.FC = () => {

  const userName = useSelector((state: any) => state.UserReducers.userName);
  return (
    <SafeAreaView style={InformationScreenStyle.safeAreaStyle}>
      <View style={InformationScreenStyle.container}>
        <Text style={InformationScreenStyle.text}> User Name:</Text>
        <Text testID="User Name" style={[InformationScreenStyle.text, InformationScreenStyle.userText]}>{userName}</Text>
      </View>
    </SafeAreaView>
  );
};

export default InformationScreen;
