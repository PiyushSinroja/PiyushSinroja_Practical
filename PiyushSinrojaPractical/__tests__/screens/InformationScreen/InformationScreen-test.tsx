import React from 'react';
import InformationScreen from '../../../source/screens/InformationScreen/InformationScreen';
import {render} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import configureStore from '../../../source/redux-classes/ConfigStore';

const store = configureStore();

describe('Information Screen', () => {
  it('renders properly', () => {
    const {toJSON, getByTestId} = render(
      <Provider store={store}>
        <InformationScreen />
      </Provider>,
    );
    const textInstance = getByTestId('User Name');
    expect(textInstance).toBeDefined();
    expect(toJSON()).toMatchSnapshot();
  });
});
