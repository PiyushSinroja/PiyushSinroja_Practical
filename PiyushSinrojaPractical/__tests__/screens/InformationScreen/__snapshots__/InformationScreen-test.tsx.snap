// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information Screen renders properly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 19,
          "height": 25,
          "marginTop": 10,
          "textAlign": "left",
        }
      }
    >
       User Name:
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 19,
            "height": 25,
            "marginTop": 10,
            "textAlign": "left",
          },
          Object {
            "marginLeft": 10,
            "marginRight": 15,
          },
        ]
      }
      testID="User Name"
    >
      
    </Text>
  </View>
</RCTSafeAreaView>
`;
